service/storage_service.cc

```diff
@@ -1088,11 +1082,24 @@ void storage_service::handle_state_normal(inet_address endpoint) {
     if (_gossiper.uses_host_id(endpoint)) {
         auto host_id = _gossiper.get_host_id(endpoint);
         auto existing = _token_metadata.get_endpoint_for_host_id(host_id);
+        auto replace_addr = db().local().get_replace_address();
         if (db().local().is_replacing() &&
-            db().local().get_replace_address() &&
-                _gossiper.get_endpoint_state_for_endpoint_ptr(db().local().get_replace_address().value())  &&
-            (host_id == _gossiper.get_host_id(db().local().get_replace_address().value()))) {
-            slogger.warn("Not updating token metadata for {} because I am replacing it", endpoint);
+            replace_addr &&
+            _gossiper.get_endpoint_state_for_endpoint_ptr(*replace_addr) &&
+            host_id == _gossiper.get_host_id(*replace_addr)) {
+            slogger.warn("Not updating token metadata for {} because I am replacing it, myip={}, node={} becomes normal status, existing node={}, node to be replaced={}",
+                    endpoint, get_broadcast_address(), endpoint, existing, *replace_addr);
+            // If the replacing node becomes NORMAL status, we can remove the
+            // node to be replaced from token_metadata. However, if the
+            // replacing node has the same ip address as the node to be
+            // replaced, we should not remove the node itself from
+            // token_metadata.
```

Comment:
The comment should explain **why** we can/cannot do something, not just state the fact that we can/cannot (which follows directly from the code).

Reply:
I mentioned we should not remove the node itself. Do you want me to explain why we can not remove the node itself from token_metadata?

Reply:
We wouldn't remove the node itself, we would remove a different node which just happens to have the same IP.
I know the tokens stay the same, but don't we have to update host IDs for example?